src/bindings/c/tests/ie_c_api_test.cpp

```diff
@@ -1369,6 +1388,9 @@ TEST(ie_infer_request_set_batch, setZeroBatch) {
 }
 
 TEST(ie_infer_request_set_batch, setNegativeBatch) {
+    #ifdef __aarch64__
```

Comment:
we must not streak the code with multiple ifdef else. Why can we make most of the tests to be generic?

Reply:
It means we should not skip tests like `ie_core_register_plugins`. Why are they not enabled for ARM? 
Tests must not rely on whether they are run on ARM or RISC-V platform, they must rely on plugin capabilities.

Reply:
@ilya-lavrenov will refactor this PR: 
1) Will create the test plugin.xml file in runtime, so the related APIs like 'ie_core_register_plugins' will enable for ARM plugin too.
2) Will move the related APIs like 'setNegativeBatch' (ARM plugin doesn't support it for now) together & use only one "#ifndef __aarch64__" to reduce multiple ifdef else using.
What do you think?

Reply:
Had finished the refactor, pls help review. Test pass on windows/linux(intel cpu) & Mac OS(ARM cpu).